### PR TITLE
Localizer should work with inner classes as well

### LINKED ISSUE #4664: Cannot Translate inner class by msgctxt
[Fully Example to this bug](https://github.com/Ahmed-Galal85/TestAppForLocalization/tree/master/App.Web/App.Web)
Inner class not working: msgctxt "App.Web.Models.HomeVM.Index"

```
#: App.Web\Models\HomeVM.cs:13
#. [Display(Name = "العنوان")]
msgctxt "App.Web.Models.HomeVM.Index"
msgid "العنوان"
msgstr "Title"
```

```
namespace App.Web.Models
{
    public static class HomeVM
    {

        public class Index
        {
            [Display(Name = "العنوان")]//not working
            public string Title { get; set; }
        }
    }
}

```
normal class working well: msgctxt "App.Web.Models.PrivacyVM"

```
msgctxt "App.Web.Models.PrivacyVM"
msgid "عنوان الخصوصية"
msgstr "Privacy Title"

namespace App.Web.Models
{
    public class PrivacyVM
    {
        [Display(Name ="عنوان الخصوصية")]
        public string TitlePrivacy { get; set; }
    }
}
```

thanks

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #4664

### CODE REVIEW

**Comment by hishamco** [comment]
Oops, seems the CI fails, it working at my side, I will rebuild the solution and try to test it again

**Comment by hishamco** [comment]
Nah this is frequently happen with Fork git client, I don't know why some fields ignored from commit occasionally

@Skrypt your final review before merge this, now i'm ready to dig into the data annotations only

**Comment by hishamco** [comment]
@Skrypt this PR is ready to merge, could please have a look and merge if you don't have any feedback

File: src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PortableObjectStringLocalizerFactory.cs [hunk 1]
```diff
@@ -60,5 +65,14 @@ public IStringLocalizer Create(string baseName, string location)
 
             return new PortableObjectStringLocalizer(relativeName, _localizationManager, _fallBackToParentCulture, _logger);
         }
+
+        // The context within inner class
+        private void TryFixInnerClassPath(ref string context)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PortableObjectStringLocalizerFactory.cs]: Can you try to just return the new string? The `ref` should not be necessary in this case (just makes it more copmlex to read). Also missing space after `if`

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PortableObjectStringLocalizerFactory.cs]: > Can you try to just return the new string?

Ok

> Also missing space after if

Good catch :)

**Comment by hishamco** [comment]
Done!!

**Pull request merged**